frontend/packages/kubevirt-plugin/src/k8s/enhancedK8sMethods/__tests__/enhancedK8sMethods.spec.ts

```diff
@@ -0,0 +1,205 @@
+    data: K8sResourceCommon,
+    patches: Patch[],
+    enhancedOpts,
+  ) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(
+      new HistoryItem(HistoryType.PATCH, { ...data, ...patches[0] }),
+      enhancedOpts,
+    );
+  };
+  methodsUnchecked.k8sKill = async (
+    kind: K8sKind,
+    data: K8sResourceCommon,
+    opts,
+    json,
+    enhancedOpts,
+  ) => {
+    methodsUnchecked.registerKind(kind);
+    methodsUnchecked.appendHistory(new HistoryItem(HistoryType.DELETE, data), enhancedOpts);
+  };
+  return { methods, methodsUnchecked };
+};
+
+const expectHistory = (history, expectedHistory) => {
+  expect(history).toHaveLength(expectedHistory.length);
+
+  expectedHistory.forEach((expectHistoryItem, idx) => {
+    expect(history[idx]).toEqual(expectHistoryItem);
+  });
+};
+
+describe('enhancedK8sMethods.js', () => {
+  const testVM = new VMWrapper()
+    .setModel(VirtualMachineModel)
+    .setName('testVM')
+    .setNamespace('default')
+    .setMemory('5', 'Gi')
+    .asResource();
+
+  const otherTestVM = new VMWrapper()
+    .setModel(VirtualMachineModel)
+    .setName('otherVM')
+    .setNamespace('kube')
+    .setMemory('1', 'Mi')
+    .asResource();
+
+  const testPatch = new PatchBuilder('/')
+    .add({
+      additionalData: 'testData',
+    })
+    .build();
+
+  it('records history and shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    expect(methods.getHistory()).toHaveLength(0);
+    expect(methods.getActualState()).toHaveLength(0);
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    const history = methods.getHistory();
+
+    expectHistory(history, [
+      new HistoryItem(HistoryType.CREATE, testVM),
+      new HistoryItem(HistoryType.PATCH, { ...testVM, ...testPatch }),
+      new HistoryItem(HistoryType.DELETE, testVM),
+    ]);
+
+    expect(methodsUnchecked.history).toHaveLength(3);
+  });
+
+  it('shows actualState', async () => {
+    const { methods, methodsUnchecked } = spyEnhancedK8sMethods();
+
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methods.k8sPatch(VirtualMachineModel, testVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sKill(VirtualMachineModel, testVM);
+    await methods.k8sCreate(VirtualMachineModel, otherTestVM);
+    await methods.k8sPatch(VirtualMachineModel, otherTestVM, []);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, otherTestVM);
+    await methods.k8sCreate(VirtualMachineModel, testVM);
+    await methodsUnchecked.k8sGet(VirtualMachineModel, testVM);
+    await methods.k8sPatch(VirtualMachineModel, testVM, [testPatch]);
+    const actualState = methods.getActualState();
+
+    expect(actualState).toHaveLength(2);
```

Comment:
can you destruct `actualState` for readability? or add a comment? I have no idea whats inside.

Reply:
added a better comment to the `getActualState` method

Reply:
and returns ` K8sResourceKind[]`